controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

arm_controller:

  ros__parameters:
    joints:
      - waist
      - shoulder
      - elbow
      - wrist_rotation
      - wrist_flex
      - hand_roll
    command_interfaces:
      - position
      - velocity

    state_interfaces:
      - position
      - velocity

    # state_publish_rate: 200.0 # Defaults to 50
    # action_monitor_rate: 20.0 # Defaults to 20

    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 1.5
      waist:
        trajectory: 0.05
        goal: 0.03
      shoulder:
        trajectory: 0.05
        goal: 0.03
      elbow:
        trajectory: 0.05
        goal: 0.03
      wrist_rotation:
        trajectory: 0.05
        goal: 0.03
      wrist_flex:
        trajectory: 0.05
        goal: 0.03
      hand_roll:
        trajectory: 0.05
        goal: 0.03

# joint_state_broadcaster:
#   ros__parameters:
#     type: joint_state_broadcaster/JointStateBroadcaster
#     joints:
#       - waist
#       - shoulder
#       - elbow
#       - wrist_rotation
#       - wrist_flex
#       - hand_roll